src/test/spec/oidc.rs

```diff
@@ -3,36 +3,650 @@ use crate::{
         auth::{oidc, AuthMechanism, Credential},
         options::ClientOptions,
     },
-    test::log_uncaptured,
+    test::{log_uncaptured, FailCommandOptions, FailPoint},
     Client,
 };
-use std::sync::{Arc, Mutex};
+use bson::{doc, Document};
+use futures_util::FutureExt;
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::sync::Mutex;
+
```

Comment:
Some of these it's sort of strange for their to be macros, but I did it for consistency with mongodb_uri_multi

Reply:
Also, note, the plan is currently for multi idp to only be tested on linux, so this makes some sense.